File: ObserverV2Java/ObserverJava/src/src/AnalogClock.java
package src;

import include.IAnalogClock;
import include.IClockTimer;

public class AnalogClock implements IAnalogClock {

    public AnalogClock(IClockTimer clockTimer) {
        timer = clockTimer;
        timer.getSubscriber().attach(this);
    }

    public void update() {
        draw();
    }


    public void draw() {
        String hours = String.valueOf(timer.getHour());
        String minutes = String.valueOf(timer.getMinute());
        String seconds = String.valueOf(timer.getSecond());

        if (hours.length() == 1) {
            hours = "0" + hours;
        }
        if (minutes.length() == 1) {
            minutes = "0" + minutes;
        }
        if (seconds.length() == 1) {
            seconds = "0" + seconds;
        }
        System.out.println("I am Analog: " + hours + ":" + minutes + ":" + seconds);
    }

    private IClockTimer timer;
}

File: ObserverV2C#/Observer/src/AnalogClock.cs
using Observer.include;
using System;
using System.Collections.Generic;
using System.Text;

namespace Observer.src
{
    public class AnalogClock: IAnalogClock {
        public AnalogClock(IClockTimer clockTimer)
        {
            timer = clockTimer;
            timer.GetSubscriber().Attach(this);
        }

        public void Update()
        {
            Draw();
        }

        public void Draw()
        {
            string hours = timer.GetHour().ToString();
            string minutes = timer.GetMinute().ToString();
            string seconds = timer.GetSecond().ToString();

            if (hours.Length == 1) { hours = "0" + hours; }
            if (minutes.Length == 1) { minutes = "0" + minutes; }
            if (seconds.Length == 1) { seconds = "0" + seconds; }
            Console.WriteLine("I am Analog: " + hours + ":" + minutes + ":" + seconds);
        }

        private IClockTimer timer;
    }
}
